StRoot/StFstRawHitMaker/StFstRawHitMaker.cxx

```diff
@@ -0,0 +1,598 @@
+				}
+				else {      // ZS data
+					signalCorrected[channel][timebin] = signalUnCorrected[channel][timebin];
+				}
+			}
+		} // end current APV loops
+
+		nIdTruth_Fst += FillRawHitCollectionFromAPVData(dataFlag, ntimebin, counterAdcPerRgroupPerEvent, sumAdcPerRgroupPerEvent, apvElecId, signalUnCorrected, signalCorrected, idTruth);
+
+	}//end while
+	LOG_INFO << " Total number of FST Raw Hits - Step I = " << mFstCollectionPtr->getNumRawHits() << " w/ idTruth = " << nIdTruth_Fst << endm;
+
+	// In case of pure simulation mode when neither real data hits from DAQ
+	// records is available nor embedding is requested fill the output container
+	// with simulated hits
+	if(!mDoEmbedding && !nRawAdcFromData) nIdTruth_Fst += FillRawHitCollectionFromSimData();
+
+	LOG_INFO << " Total number of FST Raw Hits - Step II = " << mFstCollectionPtr->getNumRawHits() << " w/ idTruth = " << nIdTruth_Fst << endm;
+
+	return ierr;
+}
+
+
+/**
+ * A private helper function to actually insert StFstRawHits unpacked from DAQ
+ * records into the final output container mFstCollectionPtr.
+ */
+int StFstRawHitMaker::FillRawHitCollectionFromAPVData(unsigned char dataFlag, int ntimebin,
+		int counterAdcPerRgroupPerEvent[][kFstNumTimeBins], double sumAdcPerRgroupPerEvent[][kFstNumTimeBins], int apvElecId,
+		std::array< std::array<double, kFstNumTimeBins>, kFstNumApvChannels > &signalUnCorrected,
+		std::array< std::array<double, kFstNumTimeBins>, kFstNumApvChannels > &signalCorrected,
+		std::array<int, kFstNumApvChannels> &idTruth)
+{
+	int nIdTruth = 0;
+	// calculate the dynamical common mode noise for the current chip in this event
+	double commonModeNoise[kFstNumRStripsPerWedge][kFstNumTimeBins];
+
+	for (int tbIdx = 0; tbIdx < kFstNumTimeBins; tbIdx++){
+		for (int rindex = 0; rindex < kFstNumRStripsPerWedge; rindex++)
+			commonModeNoise[rindex][tbIdx] = 0.;
+	}
+
+	if ( !mIsCaliMode && dataFlag == mADCdata ) {
+		for (short iTb = 0; iTb < ntimebin; iTb++)  {
+			for (int rindex = 0; rindex < kFstNumRStripsPerWedge; rindex++){
+				if (counterAdcPerRgroupPerEvent[rindex][iTb] > 0)
+					commonModeNoise[rindex][iTb] = sumAdcPerRgroupPerEvent[rindex][iTb] / counterAdcPerRgroupPerEvent[rindex][iTb];
+			}
+		}
+	}
+
+	// CMN correction, raw hit decision and channel counter passed the hit decision
+	std::vector<bool> isPassRawHitCut(kFstNumApvChannels, false);
+	Int_t nChanPassedCut = 0;
+
+	for (int iChan = 0; iChan < kFstNumApvChannels; iChan++)
+	{
+		Int_t elecId = apvElecId + iChan;
+		Int_t geoId  = mMappingVec[elecId];
+		// Modify signalCorrected values in some ways
+		for (int iTBin = 0; iTBin < ntimebin; iTBin++)
+		{
+			if (!mIsCaliMode && mDoCmnCorrection && dataFlag == mADCdata ){
+				int rstrip = (geoId % (kFstNumInnerSensorsPerWedge * kFstNumStripsPerInnerSensor + kFstNumOuterSensorsPerWedge * kFstNumStripsPerOuterSensor))/kFstNumPhiSegPerWedge;
+				for (int rindex = 0; rindex < kFstNumRStripsPerWedge; rindex++){
+					if(rindex == rstrip){
```

Comment:
Same question. Why not just access the element by using rstrip?

Reply:
fixed by using rstrip directly and removed the loop.